src/main/java/org/tikv/common/operation/iterator/ScanIterator.java

```diff
@@ -107,7 +104,7 @@ boolean cacheLoadFails() {
         startKey = lastKey.next().toByteString();
       }
       // notify last batch if lastKey is greater than or equal to endKey
-      if (hasEndKey && lastKey.compareTo(endKey) >= 0) {
+      if (hasEndKey && lastKey.compareTo(endKey) >= 0 || startKey.isEmpty()) {
```

Comment:
`startKey.isEmpty()` always equals false

Reply:
it is possible that startKey is ByteString.EMPTY, which indicates +♾

Reply:
ok, please update the comment

Reply:
@marsishandsome done